test/compress/collapse_vars.js

```diff
@@ -1363,3 +1365,44 @@ issue_1562: {
         for (; f(z + 2) ;) bar(30);
     }
 }
+
+issue_1605_1: {
+    options = {
+        collapse_vars: true,
+        toplevel: false,
+    }
+    input: {
+        function foo() {
+            return 1;
+        }
+        var o = new Object;
+        o.p = 1;
+    }
+    expect: {
+        function foo() {
+            return 1;
+        }
+        var o = new Object;
+        o.p = 1;
+    }
+}
+
+issue_1605_2: {
+    options = {
+        collapse_vars: true,
+        toplevel: "vars",
+    }
+    input: {
+        function foo() {
+            return 1;
+        }
```

Comment:
Could you please define `function foo` as the following to show that `collapse_vars` works in functions regardless of the toplevel setting?
```
function foo(x){ var y = x; return y; }
```